### PR TITLE
LXNavigation EOS/ERA/LX10k driver

### PR DESCRIPTION
**Pull request by AlphaLima**
Task declaration
MC, BAL, BUG bi-directional exchange
Read Logbook
Parse AHRS and G-Force
radio support
get Polar from device

### CODE REVIEW

File: Common/Source/Dialogs/dlgLXIGCDownload.cpp [hunk 1]
```diff
@@ -220,29 +220,28 @@ static void OnMultiSelectListPaintListItem(WindowControl *Sender, LKSurface &Sur
...
 
     TCHAR *remaining;    /* extract filname */
     TCHAR *IGCFilename = _tcstok_r(
-            LX_IGCReadDialog.FileList()->at(LX_IGCReadDialog.DrawIndex()).Line1, TEXT(" "),
+        szTmp, TEXT(" "),
             &remaining);
 
     TCHAR PathAndFilename[MAX_PATH];
     LocalPath(PathAndFilename, _T(LKD_LOGS), IGCFilename);     // add path
+    _sntprintf(text1, MAX_NMEA_LEN, _T("%s"),text1);     // missing
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgLXIGCDownload.cpp:237]: result of using same buffer for source and destination is undefined

File: Common/Source/Dialogs/dlgLXIGCDownload.cpp [hunk 2]
```diff
@@ -220,29 +220,28 @@ static void OnMultiSelectListPaintListItem(WindowControl *Sender, LKSurface &Sur
...
     if (Appearance.UTF8Pictorials)                             // use UTF8 symbols?
     {
       if (lk::filesystem::exist(PathAndFilename))                // check if file exists
-        _sntprintf(text1, MAX_NMEA_LEN, _T("✔ %s"), IGCFilename); // already copied
-      else
-        _tcscpy(text1, IGCFilename);                             // missing
+        _sntprintf(text1, MAX_NMEA_LEN, _T("✔ %s"), text1); // already copied
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgLXIGCDownload.cpp:241]: result of using same buffer for source and destination is undefined

File: Common/Source/Dialogs/dlgLXIGCDownload.cpp [hunk 3]
```diff
@@ -220,29 +220,28 @@ static void OnMultiSelectListPaintListItem(WindowControl *Sender, LKSurface &Sur
...
     } else {
       if (lk::filesystem::exist(PathAndFilename))                // check if file exists
-        _sntprintf(text1, MAX_NMEA_LEN, _T("* %s"), IGCFilename);// already copied
-      else
-        _tcscpy(text1, IGCFilename);                             // missing
+       _sntprintf(text1, MAX_NMEA_LEN, _T("* %s"), text1);// already copied
```
Comment (brunotl) [inline Common/Source/Dialogs/dlgLXIGCDownload.cpp:244]: result of using same buffer for source and destination is undefined

**Comment by AlphaLima** [comment]
done with next commit!

**Comment by brunotl** [comment]
Android build failed :

```shell
  ...\Common\Source\Devices\devLXNano3.cpp:1715:9: error: 'snprintf' size argument is too large; destination buffer has size 80, but size argument is 260 [-Werror,-Wfortify-source]
  
          _sntprintf (szPolarName ,MAX_PATH, _T("%s"), d->Name );
  ```

**Comment by brunotl** [comment]
this pull request is concerne new LX EOS/ERA driver, why it change the LX Nano3 driver ?

File: Common/Source/Dialogs/dlgLXIGCDownload.cpp (hunk 1 above)
Comment (AlphaLima) [reply Common/Source/Dialogs/dlgLXIGCDownload.cpp]: done

**Comment by AlphaLima** [comment]
OK fixed the string length and put the Nano3 changes in a seperate commit on top of the LX EOS driver.

**Comment by brunotl** [comment]
Two error :
```
Common\Source\Devices\devLX_EOS_ERA.cpp:1483:15: error: 'snprintf' size argument is too large; destination buffer has size 80, but size argument is 260 [-Werror,-Wfortify-source]
              _sntprintf (szPolarName ,MAX_PATH, _T("%s"), d->Name );

```
```
Common\Source\lk8000.cpp:518: error: undefined reference to 'DevLX_EOS_ERA::Register()'
  clang++: error: linker command failed with exit code 1 (use -v to see invocation)
  ninja: build stopped: subcommand failed.
```

**Comment by brunotl** [comment]
Fixed and merged manually

**Pull request closed without merging**